Modele_MB.R

```diff
@@ -50,32 +59,35 @@ library(treeio)
 rec_table <- list(node = seq_len(n_tips + n_nodes))
 
 # Positions des nœuds estimées
-rec_table[["location1"]] <- c(dat$lat, mut_hat[1], Z_hat[,1])
-rec_table[["location2"]] <- c(dat$long, mut_hat[2], Z_hat[,2])
+rec_table[["location1"]] <- c(dat$lat, Z_hat[,1])
+rec_table[["location2"]] <- c(dat$long, Z_hat[,2])
 
 # Ellipses de confiance
-# mut_hat = AY
-A <- solve(t(Un)%*%solve(Cn)%*%Un)%*%t(Un)%*%solve(Cn)
-# Z_hat = QY
-Q <- Um%*%A + Cmn%*%solve(Cn)%*%(diag(n)-Un%*%A)
+# # mut_hat = AY
+# A <- solve(t(Un)%*%solve(Cn)%*%Un)%*%t(Un)%*%solve(Cn)
+# # Z_hat = QY
+# Q <- Um%*%A + Cmn%*%solve(Cn)%*%(diag(n)-Un%*%A)
+# var Z | Y
+VZY <- Cm - Cmn%*%solve(Cn)%*%t(Cmn)
```

Comment:
I'm not sure I understand the previous formulas (but I did not look into them closely).
The formula for `VZY` directly derives from the standard Gaussian conditionning.